### PR TITLE
Migrate remaining design system constants to enums

### LINKED ISSUE #19130: Migrate remaning design system consts to enums
### Description
There are a few remaining design system consts that can be migrated to enums in `ui/helpers/constants/design-system.ts`

- `DISPLAY`
- `FLEX_DIRECTION`
- `FLEX_WRAP`
- `BLOCK_SIZES`
- `SEVERITIES`

### Technical Details
- Create enum versions of `DISPLAY`, `FLEX_DIRECTION`, `FLEX_WRAP`, `BLOCK_SIZES`. Use PascalCase for the name and keys and update `BLOCK_SIZES` to use singular e.g. `BlockSize`
- Add deprecation message above old constants to notify other contributors
- Update Box component storybook documentation and stories
- Point contributors to this issue in deprecation messages https://github.com/MetaMask/metamask-extension/issues/18714

### Acceptance Criteria
- enum versions exist e.g. `Display`, `FlexDirection`, `FlexWrap` and `BlockSize`
- constants have been deprecated
- Stories, documentation and tests have been updated for the `Box` component

If the acceptance criteria is not met, PRs may be closed.

### Difficulty: Intermediate

Good first issue for: External contributors who are familiar with enums and want to contribute to improving the stability of the extension

### PR DESCRIPTION
**Pull request by nazrul7711**
## Explanation
Currently there are some remaining design system constants that need to be migrated to enums. This PR creates enums for the remaining constants. It also deprecates the constants in favour of the enums and points to a good first issue to encourage contributors to replace them.

*  Enums for design system constants `DISPLAY`, `FLEX_DIRECTION`, `FLEX_WRAP`, `BLOCK_SIZES`, `SEVERITIES` have been created
* `Box` component, documentation and tests updated to favour enums but support constants through unit testing
* Deprecation JSDoc notices added so strike through appears in VS Code as well as good first issue to replace deprecated constants https://github.com/MetaMask/metamask-extension/issues/18714 (Will add constants in the PR once it's merged)
* Fixes: #19130

### Before

https://github.com/MetaMask/metamask-extension/assets/8112138/b4bebdcf-945f-4597-87d7-b28686bf803a

### After

https://github.com/MetaMask/metamask-extension/assets/8112138/03799cc4-a6ba-4782-85fd-c5792b90e59d

## Manual Testing Steps

- Pull this branch
- Do a search of the code base for any of the deprecated constants `DISPLAY`, `FLEX_DIRECTION`, `FLEX_WRAP`, `BLOCK_SIZES`, `SEVERITIES`
- Ensure you have the correct VS Code settings enabled for deprecation indicators
- Check that each constant is deprecated according to JSDoc format

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [x] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [x] PR is linked to the appropriate GitHub issue
- [x] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

**Review by georgewrmarshall — CHANGES_REQUESTED** [review]
Hey @nazrul7711, thanks for your contribution. This is looking really great! I've left a few small suggestions. It would also be great if you could update the PR description according to the template provided. This really helps with review.

I wasn't sure if we would have to update the `box.d.ts` file with a union type of the `Display` enum as well as the `DISPLAY` const? There doesn't seem to be any errors in the TypeScript files I was looking at which I found interesting I suppose because it is just looking for the string or an array of strings.

File: yarn.lock
[the remark this answers is not in the document]
Comment (nazrul7711) [reply yarn.lock:1]: ok

File: ui/helpers/constants/design-system.ts [hunk 1]
```diff
@@ -235,19 +235,61 @@ export enum JustifyContent {
...
+  Flex = 'flex',
+  Grid = 'grid',
+  InlineBlock = 'inline-block',
+  Inline = 'inline',
+  InlineFlex = 'inline-flex',
+  InlineGrid = 'inline-grid',
+  ListItem = 'list-item',
+  None = 'none',
+}
+/**
```
[the remark this answers is not in the document]
Comment (nazrul7711) [reply ui/helpers/constants/design-system.ts:287]: added

File: ui/helpers/constants/design-system.ts [hunk 2]
```diff
@@ -289,6 +331,45 @@ export const FRACTIONS = {
...
+
+/**
+ * @deprecated `BLOCK_SIZES` const has been deprecated in favour of the `BlockSize` enum which can still be imported from `ui/helpers/constants/design-system.ts`
+ *
+ * Help to replace `BLOCK_SIZES` with `BlockSize` by submitting PRs against https://github.com/MetaMask/metamask-extension/issues/18714
+ */
+
 export const BLOCK_SIZES = {
   ...FRACTIONS,
   SCREEN: 'screen',
```
[the remark this answers is not in the document]
Comment (nazrul7711) [reply ui/helpers/constants/design-system.ts:375]: depracated

File: ui/components/ui/box/box.js [hunk 3]
```diff
@@ -212,7 +212,7 @@ const Box = React.forwardRef(function Box(
     alignItems,
     justifyContent,
     textAlign,
-    flexDirection = FLEX_DIRECTION.ROW,
+    flexDirection = FlexDirection.ROW,
```
[the remark this answers is not in the document]
Comment (nazrul7711) [reply ui/components/ui/box/box.js:215]: rectified

File: yarn.lock
[the remark this answers is not in the document]
Comment (nazrul7711) [reply yarn.lock:1]: removed

**Comment by nazrul7711** [comment]
hi @georgewrmarshall I have made the modifications , kindly review it.

File: yarn.lock
Comment (georgewrmarshall) [inline yarn.lock:1]: Sorry if I was unclear, I didn't mean to remove the yarn.lock file just revert the changes so it doesn't appear in your PR

Comment (nazrul7711) [reply yarn.lock:1]: so u mean i should have yarn.lock in my repo but remove it from the PR.rest is fine?

**Review by georgewrmarshall — CHANGES_REQUESTED** [review]
Hey @nazrul7711, I've updated the PR title and description so the PR is a little more clear. Some of it was still commented out. I've also realized that I missed `SEVERITIES` in the issue(now updated). Could you also create an enum for `Severity`. I don't think we need one for `RESIZE` as it's used for a single component that will likely be migrated to TypeScript and have it's own types.

File: ui/components/component-library/tag-url/tag-url.js [hunk 4]
```diff
@@ -38,7 +38,7 @@ export const TagUrl = ({
       paddingRight={4}
       gap={2}
       borderRadius={BorderRadius.pill}
-      display={DISPLAY.FLEX}
+      display={Display.Flex}
       {...props}
     >
       <AvatarFavicon src={src} name={label} {...avatarFaviconProps} />
```
[the remark this answers is not in the document]
Comment (nazrul7711) [reply ui/components/component-library/tag-url/tag-url.js:44]: modified

**Review by georgewrmarshall — APPROVED** [review]
Hey @nazrul7711, I rebased this PR to remove the `yarn.lock` changes and made sure that the deprecated constants still work by keeping the unit tests for them. This was kind of blocking some other TS migration work so I really appreciate your timing on this and hope you don't mind my updates. LGMT 🔥

**Review by brad-decker — APPROVED** [review]

**Pull request merged**